summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -183,6 +197,12 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
     }
 
     val countersForSpout: Seq[(Group, Name)] = JobCounters.getCountersForJob(jobID).getOrElse(Nil)
+    val isMergeableWithSource = getOrElse(stormDag, node, DEFAULT_FM_MERGEABLE_WITH_SOURCE).get
+    val sourceParallelism = getOrElse(stormDag, node, parOpt.getOrElse(DEFAULT_SOURCE_PARALLELISM)).parHint
+    val FMParallelism = getOrElse(stormDag, node, DEFAULT_FM_PARALLELISM).parHint
+
+    if (isMergeableWithSource && (FMParallelism > sourceParallelism))
```

Comment:
I think this is better as a require.  Like:

if (isMergeableWithSource) {
  require(FMParallelism <= sourceParallelism, s"SourceParallelism ($sourceParallelism) must be at least as high as FlatMapParallelism ($flatMapParallelism) when merging flatMap with Source")
}

Note that this will cause a `IllegalArgumentException` instead of a `RuntimeException`.